lib/shark.rb

```diff
@@ -34,26 +34,45 @@ def self.configure
   # @api public
   def self.with_service_token(token)
     if token.is_a?(String)
-      self.service_token = token
+      self.auth_token = "Bearer #{token}"
     elsif token.respond_to?(:jwt)
-      self.service_token = token.jwt
+      self.auth_token = "Bearer #{token.jwt}"
     else
       raise ArgumentError, 'Parameter :token must be kind of String.'
     end
 
     yield
   ensure
-    self.service_token = nil
+    self.auth_token = nil
+  end
+
+  # Within the given block, add the authorization header token to all api requests.
+  #
+  # @param token [String] The token for the authorization header
+  # @param block [Block] The block where authorization token will be set for
+  # @api public
+  def self.with_auth_token(token)
+    if token.is_a?(String)
+      self.auth_token = token
+    elsif token.respond_to?(:jwt)
+      self.auth_token = token.jwt
+    else
+      raise ArgumentError, 'Parameter :token must be kind of String.'
+    end
```

Comment:
Would it not be better / simpler to only allow strings?
What case do you have in mind for `elsif token.respond_to?(:jwt)`?

Reply:
it was a copy of your code (see `with_service_token`), I thought you know :)

Reply:
I feel I understood, it is for tokens stored in DB. Removing it.